pkg/providers/vsphere/setupuser/setupuser.go

```diff
@@ -0,0 +1,187 @@
+package setupuser
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/aws/eks-anywhere/pkg/config"
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/providers/vsphere"
+)
+
+const (
+	vSphereRootPath = "/"
+)
+
+func SetupGOVCEnv(ctx context.Context, vsuc *VSphereSetupUserConfig) error {
+	os.Setenv("GOVC_URL", vsuc.Spec.Connection.Server)
+	os.Setenv("GOVC_INSECURE", strconv.FormatBool(vsuc.Spec.Connection.Insecure))
+	os.Setenv("GOVC_DATACENTER", vsuc.Spec.Datacenter)
+	return nil
+}
+
+func Run(ctx context.Context, vsuc *VSphereSetupUserConfig, govc vsphere.ProviderGovcClient) error {
```

Comment:
I would avoid depending on the `ProviderGovcClient` (bad naming too) and define the interface you need in this package.